util.go

```diff
@@ -1,11 +1,38 @@
 package tuna
 
 import (
+	"errors"
+	"io/ioutil"
+	"net/http"
 	"strings"
 
 	"github.com/nknorg/nkn/common"
 )
 
+const IP2CUrl = "https://ip2c.org/"
+
+type Location struct {
+	Code      string  `json:"Code"`
+	Country   string  `json:"Country"`
+	City      string  `json:"City"`
+	Latitude  float32 `json:"Latitude"`
+	Longitude float32 `json:"Longitude"`
+}
+
+func (l *Location) Empty() bool {
+	return *l == Location{}
```

Comment:
A tiny improvement: define a variable `var emptyLocation = Location{}`, and then compare `*l == emptyLocation` to avoid unnecessary memory allocation for each `Empty()` call.